frappe/database/postgres/database.py

```diff
@@ -416,6 +416,34 @@ def get_column_type(self, doctype, column):
 	def get_database_list(self):
 		return self.sql("SELECT datname FROM pg_database", pluck=True)
 
+	def get_column_index(
+		self, table_name: str, fieldname: str, unique: bool = False, order: int = 1
+	) -> frappe._dict | None:
+		"""
+		Check if column exists for a specific fields.
+		TODO: in a specific order
+		"""
+		unique_condition = "AND indexdef LIKE 'UNIQUE'" if unique else ""
+		indexes = self.sql(
+			f"""
+			SELECT
+				indexname as name,
+				indexdef as definition
+			FROM pg_indexes
+			WHERE tablename = '{table_name}'
+				AND indexname = '{fieldname}'
+				{unique_condition}
+			"""
+		)
+
+		# TODO: check specific order, now it just checks if its single column index
+		for index in indexes:
+			# in case of single column index, indexname is same as fieldname
+			is_single_column_index = index.name == fieldname
```

Comment:
This seems incorrect. 

We add suffix here: https://github.com/frappe/frappe/blob/0d5f28d569a5cac4bef92648c565a773e7c70a73/frappe/database/database.py#L1138-L1142

The function shouldn't rely on name of index at all IMO. That was the problem with previous mariadb implementation too. 


https://github.com/frappe/frappe/pull/18262